### PR TITLE
added support for job to have pause_status

### PR DESCRIPTION
**Pull request by stikkireddy**
Changed pause_status from the schedule config block in jobs to a optional+computed field rather than a computed field (least amount of disruption to existing state). 
(Users can choose to omit it if the do not want terraform to nitpick that it has changed. Changing it to just optional will give the behavior of always tracking state and always must be managed by terraform.)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@
 * Fix incorrect escaping of notebook names ([#566](https://github.com/databrickslabs/terraform-provider-databricks/pull/566))
 * Added support for spot instances on Azure ([#571](https://github.com/databrickslabs/terraform-provider-databricks/pull/571))
 
+**Behavior changes**
```
Comment (nfx) [inline CHANGELOG.md:8]: This is not a behavior change, just field that is exposed now

File: compute/model.go [hunk 2]
```diff
@@ -520,7 +520,7 @@ type JobEmailNotifications struct {
 type CronSchedule struct {
 	QuartzCronExpression string `json:"quartz_cron_expression"`
 	TimezoneID           string `json:"timezone_id"`
-	PauseStatus          string `json:"pause_status,omitempty" tf:"computed"`
```
Comment (nfx) [inline compute/model.go:523]: What happens if you keep tf computed flag? I'd rather keep the schema compatible

File: compute/resource_job.go [hunk 3]
```diff
@@ -115,6 +115,14 @@ var jobSchema = common.StructToSchema(JobSettings{},
 			p.Required = false
 		}
 
+		if p, err := common.SchemaPath(s, "schedule", "pause_status"); err == nil {
+			p.Optional = true
```
Comment (nfx) [inline compute/resource_job.go:119]: Omitempty is setting optional, so this line is not needed

File: compute/resource_job.go [hunk 4]
```diff
@@ -115,6 +115,14 @@ var jobSchema = common.StructToSchema(JobSettings{},
...
+			p.Computed = true
+			p.Type = schema.TypeString
+			p.Required = false
```
Comment (nfx) [inline compute/resource_job.go:122]: Boolean is false by default so this line is redundant

File: compute/resource_job.go [hunk 5]
```diff
@@ -115,6 +115,14 @@ var jobSchema = common.StructToSchema(JobSettings{},
 			p.Required = false
 		}
 
+		if p, err := common.SchemaPath(s, "schedule", "pause_status"); err == nil {
+			p.Optional = true
+			p.Computed = true
```
Comment (nfx) [inline compute/resource_job.go:120]: Tf:computed sets it

File: compute/resource_job.go [hunk 6]
```diff
@@ -115,6 +115,14 @@ var jobSchema = common.StructToSchema(JobSettings{},
...
+			p.Type = schema.TypeString
```
Comment (nfx) [inline compute/resource_job.go:121]: Type is discovered by reflect_resource already

**Review by nfx — CHANGES_REQUESTED** [review]
Minor edits required

File: CHANGELOG.md (hunk 1 above)
Comment (stikkireddy) [reply CHANGELOG.md]: changed it to just a line entry

File: docs/resources/job.md [hunk 7]
```diff
@@ -69,6 +69,7 @@ The following arguments are required:
 
 * `quartz_cron_expression` - (Required) A [Cron expression using Quartz syntax](http://www.quartz-scheduler.org/documentation/quartz-2.3.0/tutorials/crontrigger.html) that describes the schedule for a job. This field is required.
 * `timezone_id` - (Required) A Java timezone ID. The schedule for a job will be resolved with respect to this timezone. See Java TimeZone for details. This field is required.
+* `pause_status` - (Optional) Indicate whether this schedule is paused or not. Either “PAUSED” or “UNPAUSED”.
```
Comment (alexott) [inline docs/resources/job.md:72]: maybe write `UNPAUSED (default)`?

**Review by alexott — APPROVED** [review]
just minor comment

File: docs/resources/job.md (hunk 7 above)
Comment (nfx) [reply docs/resources/job.md:72]: is it default, though?

**Review by nfx — APPROVED** [review]

File: docs/resources/job.md (hunk 7 above)
Comment (stikkireddy) [reply docs/resources/job.md:72]: hmm... @alexott i dont think we can provide a default if the field itself is computed. So in theory we dont provide a value when we make the rest call. So it is what ever our rest api chooses to do when that field is not omitted. And our rest api docs dont seem to indicate that there is a default. Should I check with them to verify if there is a built in default?

Comment (alexott) [reply docs/resources/job.md:72]: maybe I incorrectly formulated - we need to say that this specific will be used if it's omitted. Of course it's enforced on API level... but that was just a comment

Comment (stikkireddy) [reply docs/resources/job.md:72]: ahh gotcha i can add the create behavior if it is omitted.

File: compute/model.go (hunk 2 above)
Comment (stikkireddy) [reply compute/model.go]: removed

File: docs/resources/job.md (hunk 7 above)
Comment (stikkireddy) [reply docs/resources/job.md]: added this: `When the pause_status field is omitted and a schedule is provided, the server will default to using "UNPAUSED" as a value for pause_status.`

**Review by alexott — APPROVED** [review]

**Pull request merged**